packages/core/examples/numericStepperExample.tsx

```diff
@@ -0,0 +1,247 @@
+
+                    min={MIN_VALUES[this.state.minValueIndex].value}
+                    max={MAX_VALUES[this.state.maxValueIndex].value}
+
+                    stepSize={STEP_SIZES[this.state.stepSizeIndex].value}
+                    majorStepSize={MAJOR_STEP_SIZES[this.state.majorStepSizeIndex].value}
+                    minorStepSize={MINOR_STEP_SIZES[this.state.minorStepSizeIndex].value}
+
+                    disabled={this.state.showDisabled}
+                    readOnly={this.state.showReadOnly}
+                    leftIconName={this.state.showLeftIcon ? SAMPLE_LEFT_ICON : null }
+                    placeholder={this.state.showPlaceholder ? SAMPLE_PLACEHOLDER : null }
+
+                    onUpdate={this.handleUpdate}
+                    value={this.state.value}
+                />
+                <br />
+                <div>
+                    <strong>Value:</strong> <code>{this.state.value || "null"}</code>
+                </div>
+            </div>
+        );
+    }
+
+    private renderMinValueMenu() {
+        const { minValueIndex } = this.state;
+        return this.renderSelectMenu("Minimum value", minValueIndex, MIN_VALUES, this.handleMinValueChange);
+    }
+
+    private renderMaxValueMenu() {
+        const { maxValueIndex } = this.state;
+        return this.renderSelectMenu("Maximum value", maxValueIndex, MAX_VALUES, this.handleMaxValueChange);
+    }
+
+    private renderButtonPositionMenu() {
+        const { buttonPositionIndex } = this.state;
+        return this.renderSelectMenu(
+            "Button position", buttonPositionIndex, BUTTON_POSITIONS, this.handleButtonPositionChange);
+    }
+
+    private renderIntentMenu() {
+        return <IntentSelect intent={this.state.intent} key="intent" onChange={this.handleIntentChange} />;
+    }
+
+    private renderDisabledSwitch() {
+        return this.renderSwitch("Disabled", this.state.showDisabled, this.toggleDisabled);
+    }
+
+    private renderReadOnlySwitch() {
+        return this.renderSwitch("Read-only", this.state.showReadOnly, this.toggleReadOnly);
+    }
+
+    private renderLeftIconSwitch() {
+        return this.renderSwitch("Left icon", this.state.showLeftIcon, this.toggleLeftIcon);
+    }
+
+    private renderPlaceholderSwitch() {
+        return this.renderSwitch("Placeholder text", this.state.showPlaceholder, this.togglePlaceholder);
+    }
+
+    private renderSwitch(label: string, checked: boolean, onChange: React.FormEventHandler<HTMLElement>) {
+       const key = this.labelToKey(label);
+       return (
+            <Switch
+                checked={checked}
+                label={label}
+                key={key}
+                onChange={onChange}
+            />
+        );
+    }
+
+    private renderSelectMenu(
+        label: string,
+        selectedValue: number | string,
+        options: any[],
+        onChange: React.FormEventHandler<HTMLElement>
+    ) {
+        const key = this.labelToKey(label);
```

Comment:
Ack. But again here, I think it's clearer to have a small, descriptively named helper, even if it's only used once. Maybe I can afford to be less clever with the dot-chaining though, and just use descriptively named intermediate variables in this scope (`labelTokens`, then `lowercasedLabelTokens`). Don't even really need to handle the hyphenated-token case, so we could remove that bit of logic altogether.